src/Prima/Pyxis/Container.elm

```diff
@@ -1,133 +1,672 @@
+        |> List.append (H.maybeSingleton options.title)
+        |> List.append (List.map sizeModifierToClass options.sizeModifiers)
+        |> List.append options.styles
+
+
+{-| Internal.
+Converts FlowType to helper class
+-}
+flowTypeToClass : FlowType -> Html.Attribute msg
+flowTypeToClass flowType =
+    case flowType of
+        DirectionRow ->
+            HtmlAttributes.class "directionRow"
+
+        DirectionColumn ->
+            HtmlAttributes.class "directionColumn"
 
 
-{-| Renders a `fluid` Container
+{-| Internal.
+Converts Size to class
 -}
-fluid : List (Html msg) -> Html msg
-fluid =
-    div
-        [ class "a-containerFluid directionColumn" ]
+containerSizeToClass : Size -> Html.Attribute msg
+containerSizeToClass containerSize =
+    case containerSize of
+        Regular ->
+            HtmlAttributes.class "a-container"
 
+        Fluid ->
+            HtmlAttributes.class "a-containerFluid"
 
-{-| Renders a `fluid` Container but with some `Modifier`(s) to manage it's appearance in specific Breakpoints (BP)
+
+{-| Internal.
+Converts SizeModifier to class
 -}
-fluidWithModifiers : List Modifier -> List (Html msg) -> Html msg
-fluidWithModifiers modifiers =
-    div
-        [ class <| String.join " " <| "a-containerFluid" :: List.map modifierToString modifiers
-        ]
+sizeModifierToClass : SizeModifier -> Html.Attribute msg
+sizeModifierToClass sizeModifier =
+    case sizeModifier of
+        RegularOn breakPoint ->
+            HtmlAttributes.class <| (++) "a-containerOnBp" <| breakPointToClassSuffix breakPoint
+
+        FluidOn breakPoint ->
+            HtmlAttributes.class <| (++) "a-containerFluidOnBp" <| breakPointToClassSuffix breakPoint
 
 
-{-| Renders a completely customizable container.
+{-| Internal.
+Converts breakpoint to a size suffix, used by \`sizeModifierToClass
+-}
+breakPointToClassSuffix : BreakPoint -> String
+breakPointToClassSuffix breakPoint =
+    case breakPoint of
+        Small ->
+            "Small"
+
+        Medium ->
+            "Medium"
+
+        Large ->
+            "Large"
 
-    Container.custom [ "my-custom-class", "a-container" ] [ ColumnDirection, FluidOnBpXSmall ] [ text "Lorem ipsum dolor sit amet." ]
+        XLarge ->
+            "Xlarge"
 
+
+{-| Internal.
+Helper function to update OptionsRecord of a Container Config
 -}
-custom : List String -> List Modifier -> List (Html msg) -> Html msg
-custom classes modifiers =
-    div
-        [ class <| String.join " " <| classes ++ List.map modifierToString modifiers
-        ]
+updateOptionsRecord : OptionsRecord msg -> Config msg -> Config msg
+updateOptionsRecord optionsRecord config =
+    let
+        configRecord =
+            deObfuscateConfig config
+    in
+    Config { configRecord | options = Options optionsRecord }
+
+
+{-| Configuration modifier. Adds withAccessKey attribute on the container
+-}
+withAccessKey : Char -> Config msg -> Config msg
+withAccessKey key config =
+    let
+        configRecord =
```

Comment:
Sempre meglio passare una funzione piuttosto che leggere così da config. In questo modo si rimuovono tutti i `let .. in`

```
updateOptionsRecord
        (\c -> { c
            | accessKey = Just <| HtmlAttributes.accesskey key
        })
        config
```
Anche la nomenclatura si potrebbe rivedere:  non metterei la desinenza `Record` secondo me.

Reply:
rifattorizzato con Options (tipo opaco) e ContainerOptions per il record come per Config